functions/Get-DbaAgentServer.ps1

```diff
@@ -67,7 +67,7 @@ function Get-DbaAgentServer {
             Add-Member -Force -InputObject $jobServer -MemberType NoteProperty -Name ComputerName -Value $jobServer.Parent.ComputerName
             Add-Member -Force -InputObject $jobServer -MemberType NoteProperty -Name InstanceName -value $jobServer.Parent.ServiceName
             Add-Member -Force -InputObject $jobServer -MemberType NoteProperty -Name SqlInstance -Value $jobServer.Parent.DomainInstanceName
-            Add-Member -Force -InputObject $jobServer -MemberType ScriptProperty -Name JobHistoryIsEnabled -Value { switch ( $jobServer.MaximumHistoryRows ) { -1 { $false } default { $true } } }
+            Add-Member -Force -InputObject $jobServer -MemberType ScriptProperty -Name JobHistoryIsEnabled -Value { switch ( $this.MaximumHistoryRows ) { -1 { $false } default { $true } } }
```

Comment:
What purpose does this serve?

Reply:
Was the original method providing the wrong output?

Reply:
Yes, it was returning the wrong output. I encountered this when writing the integration test to check the JobHistoryIsEnabled property. $jobServer only has scope within this function since it is being used to create the input object for Select-DefaultView. When the ScriptProperty JobHistoryIsEnabled is accessed in other scopes (such as a pester test) the script logic executes but $jobServer as a named variable is not in scope and so the value is null. The [automatic variable $this](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables#this) allows for the ScriptProperty to correctly run in any scope.